src/components/sections/user-sync-test/user-sync-test-container.js

```diff
@@ -0,0 +1,572 @@
+    }
+
+    state = {
+        username: '',
+        runTest: false,
+        loading: true,
+        disabled: true,
+        loadData: false,
+        errorUsername: false,
+        organisationUnitsNumber: 0,
+        organisationUnitSearchNumber: 0,
+        datasetNumber: 0,
+        programNumber: 0,
+        programRuleNumber: 0,
+        metadataSize: 0,
+        dataSize: 0,
+        maxValueOUCapture: undefined,
+        maxValueOUSearch: undefined,
+        maxValueDataSet: undefined,
+        maxValueProgram: undefined,
+        maxValueProgramRule: undefined,
+        maxValueMetadata: undefined,
+        maxValueData: undefined,
+        orgUnitLoad: false,
+        dataSetLoad: false,
+        programLoad: false,
+        programRuleLoad: false,
+        metadataLoad: false,
+        dataLoad: false,
+    }
+
+    updateRecommendedValue = () => {
+        testAndroidDataConstants.forEach(test => {
+            this.setState({
+                [test.maxValueState]: test.maxValue,
+            })
+        })
+    }
+
+    clearFields = () => {
+        this.userSelected = undefined
+        this.userSelectedId = undefined
+        this.organisationUnitsNumber = 0
+        this.organisationUnitSearchNumber = 0
+        this.datasetNumber = 0
+        this.programNumber = 0
+        this.programRuleNumber = 0
+
+        this.setState({
+            userSelected: undefined,
+            userSelectedId: undefined,
+            organisationUnitsNumber: 0,
+            organisationUnitSearchNumber: 0,
+            datasetNumber: 0,
+            programNumber: 0,
+            programRuleNumber: 0,
+            metadataSize: 0,
+            dataSize: 0,
+            loadData: false,
+            orgUnitLoad: false,
+            dataSetLoad: false,
+            programLoad: false,
+            programRuleLoad: false,
+            metadataLoad: false,
+            dataLoad: false,
+        })
+    }
+
+    checkAccess = (elementArray, accessIDList) => {
+        elementArray.forEach(element => {
+            switch (element.publicAccess) {
+                case 'r-------':
+                    // if user userGroupAccess, if user userAccess
+                    break
+                case 'rw------':
+                    accessIDList.push(element)
+                    break
+                case 'rwr-----':
+                    accessIDList.push(element)
+                    break
+                case 'rwrw----':
+                    accessIDList.push(element)
+                    break
+                case 'rwrwr---':
+                    accessIDList.push(element)
+                    break
+                case 'rwrwrw--':
+                    accessIDList.push(element)
+                    break
+                case 'rwrwrwr-':
+                    accessIDList.push(element)
+                    break
+                case 'rwrwrwrw':
+                    accessIDList.push(element)
+                    break
```

Comment:
```
elementArray.forEach(element => {
            if(element.publicAccess.includes('rw')) {
                accessIDList.push(element)
            }
        })

        return accessIDList
```

Reply:
Hi @sharmyn28 - this isn’t actually quite right - `rw———` is not the same as `—rw——` so you need to check that THE FIRST TWO characters are ’rw’ if you are checking for data access (rather than metadata access).  Confusing and undocumented unfortunately....  I’m also not sure this logic is completely correct, I will open an issue after you fix this one to check for the first two characters.

Reply:
@Sharmyn28 we can get this merged for now, but please look at my response above and make sure we fix this soon - the logic is not correct here.

Reply:
See #31